FreeRTOS/FreeRTOS-Plus-TCP
# Fix network down up process

Description
-----------
I couldn't find a lot of information about the process of network going down (only some of the ports even support it) but from what I understand, it should work like this:
1. When the port detects that the link is down, it informs the network stack by calling `FreeRTOS_NetworkDown` or by sending `eNetworkDownEvent` to IP task.
2. The network stack stops any ongoing DHCP or RA process, clears ARP, etc
3. The `pfInitialize` function is periodically polled by the network stack until it returns pdPASS which signals that the link went up again.
4. The DHCP/RA processes are restarted.

During port development I noticed 2 problems:
1. The DHCP/RA timers are not stopped when the network goes down. 
2. If all interfaces are up at some point, the network stack will not poll the `pfInitialize` function when an interface goes down as the `xAllNetworksUp` flag is set and network timer is not longer checked.

In this PR I fixed 1. by disabling the DHCP_RA timer for all endpoints when the network goes down. I also fixed 2. by removing `xAllNetworksUp` altogether as I didn't have any better idea. Let me know what you think.

Test Steps
-----------
Use a port that support network down event and try disconnecting it from the network and reconnecting it. If the port is implemented correctly, the DHCP process should be stopped when the link goes down and restarted when the link goes up again.

Checklist:
----------




By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
